File: python-midonetclient/src/midonetclient/bgp_peer.py
# Copyright 2015 Midokura SARL
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may
# not use this file except in compliance with the License. You may obtain
# a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations
# under the License.

from midonetclient import resource_base
from midonetclient import vendor_media_type


class BgpPeer(resource_base.ResourceBase):

    media_type = vendor_media_type.APPLICATION_BGP_PEER_JSON

    def __init__(self, uri, dto, auth):
        super(BgpPeer, self).__init__(uri, dto, auth)

    def get_id(self):
        return self.dto['id']

    def get_asn(self):
        return self.dto['asNumber']

    def get_address(self):
        return self.dto['address']

    def get_keep_alive(self):
        return self.dto['keepAlive']

    def get_hold_time(self):
        return self.dto['holdTime']

    def get_connect_retry(self):
        return self.dto['connectRetry']

    def get_password(self):
        return self.dto['password']

    def asn(self, asn):
        self.dto['asNumber'] = asn
        return self

    def address(self, address):
        self.dto['address'] = address
        return self

    def keep_alive(self, keep_alive):
        self.dto['keepAlive'] = keep_alive
        return self

    def hold_time(self, hold_time):
        self.dto['holdTime'] = hold_time
        return self

    def connect_retry(self, connect_retry):
        self.dto['connectRetry'] = connect_retry
        return self

    def password(self, password):
        self.dto['password'] = password
        return self

File: midonet-cluster/src/main/java/org/midonet/cluster/rest_api/models/BgpPeer.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.cluster.rest_api.models;

import java.net.URI;
import java.util.UUID;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;

import com.fasterxml.jackson.annotation.JsonIgnore;

import org.midonet.cluster.data.ZoomClass;
import org.midonet.cluster.data.ZoomField;
import org.midonet.cluster.models.Topology;
import org.midonet.cluster.rest_api.ResourceUris;
import org.midonet.cluster.util.IPAddressUtil;
import org.midonet.packets.IPv4;

@ZoomClass(clazz = Topology.BgpPeer.class)
public class BgpPeer extends UriResource {

    @ZoomField(name = "id")
    public UUID id;

    @ZoomField(name = "as_number")
    @Min(1)
    public int asNumber;

    @ZoomField(name = "address", converter = IPAddressUtil.Converter.class)
    @NotNull
    @Pattern(regexp = IPv4.regex)
    public String address;

    @ZoomField(name = "keep_alive")
    @Min(5)
    @Max(3600)
    public Integer keepAlive;

    @ZoomField(name = "hold_time")
    @Min(5)
    @Max(7200)
    public Integer holdTime;

    @ZoomField(name = "connect_retry")
    @Min(5)
    @Max(3600)
    public Integer connectRetry;

    @ZoomField(name = "password")
    public String password;

    @JsonIgnore
    @ZoomField(name = "router_id")
    public UUID routerId;

    @Override
    public URI getUri() {
        return absoluteUri(ResourceUris.BGP_PEERS(), id);
    }

    public URI getRouter() {
        return absoluteUri(ResourceUris.ROUTERS(), routerId);
    }

    @Override
    @JsonIgnore
    public void create() {
        if (null == id) {
            id = UUID.randomUUID();
        }
    }

    @JsonIgnore
    public void create(UUID routerId) {
        create();
        this.routerId = routerId;
    }

    @JsonIgnore
    public void update(BgpPeer from) {
        id = from.id;
        routerId = from.routerId;
    }

}
